### PR TITLE
Revert inplace merge

### PR DESCRIPTION
**Pull request by zhanglingzhe0820**

### CODE REVIEW

**Review by Alima777 — COMMENTED** [review]
Add your PR description please~

File: server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeFileTask.java [hunk 1]
```diff
@@ -225,4 +347,13 @@ private long writeUnmergedChunks(List<Long> chunkStartTimes,
     }
     return maxVersion;
   }
-}
+
+  private File getNextMergeVersionFile(File seqFile) {
```
Comment (EJTTianYu) [inline server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeFileTask.java:351]: remove the method as it is never used

File: server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeFileTask.java [hunk 2]
```diff
@@ -195,7 +318,6 @@ private void moveUnmergedToNew(TsFileResource seqFile) throws IOException {
         ChunkCache.getInstance().clear();
         ChunkMetadataCache.getInstance().clear();
         TimeSeriesMetadataCache.getInstance().clear();
-        FileReaderManager.getInstance().closeFileAndRemoveReader(seqFile.getTsFilePath());
```
Comment (EJTTianYu) [inline server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeFileTask.java:198]: it is necessary as the old tsfile's cache is invalid

File: server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeFileTask.java [hunk 3]
```diff
@@ -19,11 +19,17 @@
...
+import static org.apache.iotdb.tsfile.common.constant.TsFileConstant.TSFILE_SUFFIX;
+
 import java.io.File;
 import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
+import java.util.Set;
```
Comment (EJTTianYu) [inline server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeFileTask.java:31]: unused import can be deleted by idea using ```control+option+O```

File: server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeFileTask.java (hunk 1 above)
Comment (zhanglingzhe0820) [reply server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeFileTask.java]: okay, I will fix it

File: server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeFileTask.java (hunk 2 above)
Comment (zhanglingzhe0820) [reply server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeFileTask.java:198]: okay, I will fix it

File: server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeFileTask.java (hunk 3 above)
Comment (zhanglingzhe0820) [reply server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeFileTask.java]: okay, I will fix it

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/level/LevelCompactionTsFileManagement.java [hunk 4]
```diff
@@ -498,6 +498,7 @@ private void merge(List<List<TsFileResource>> mergeResources, boolean sequence,
         return;
       }
     }
+    isSeqMerging = true;
```
Comment (EJTTianYu) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/level/LevelCompactionTsFileManagement.java:501]: there will be a deadlock problem here when ```isSeqMerging = true``` and unseq merge is called in line 509

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/TsFileManagement.java [hunk 5]
```diff
@@ -197,6 +198,16 @@ public synchronized void merge(boolean fullMerge, List<TsFileResource> seqMergeL
       }
       return;
     }
+    // wait until seq merge has finished
```
Comment (EJTTianYu) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/TsFileManagement.java:201]: look at the deadLock problem at LevelCompactionTsFileManagement

File: server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeFileTask.java (hunk 2 above)
Comment (EJTTianYu) [reply server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeFileTask.java:198]: add this line to your changed code.

File: server/src/test/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessorTest.java [hunk 6]
```diff
@@ -582,6 +582,11 @@ public void testMerge()
     }
 
     processor.syncCloseAllWorkingTsFileProcessors();
+    try {
+      Thread.sleep(1000);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+    }
```
Comment (EJTTianYu) [inline server/src/test/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessorTest.java:589]: using a sleep method to control the CI may be not a good idea...

**Comment by EJTTianYu** [comment]
From my point of view, using this solution to control UT is a bit tricky. The UT is ```IoTDBMergeTest.java ```
If it's just a CI problem, then it's approve here

**Review by EJTTianYu — APPROVED** [review]

**Pull request merged**